FranciscoOestmann/Topic3/src/shopping/cart/main/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,77 @@
+package shopping.cart.main;
+
+import java.util.Iterator;
+import java.util.LinkedList;
+
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+	
+	//State
+	private LinkedList<Item> list;
+	
+	//Constructor
+	public ShoppingCartServiceImpl() {
+	
+		list = new LinkedList<Item>();
+	}
+	
+	
+	//Behavior
+	
+	/**
+	 * The Method takes an element and add it to the ShopingCart list
+	 * 
+	 */
+	@Override
+	public void AddItemToCart(Item item) {
+		list.add(item);
+	}
+	/**
+	 * This Method takes an element and remove it to the ShopingCart list
+	 * 
+	 */
+	@Override
+	public  void removeItemFromCart(Item item) {
+		list.remove(item);
+		
+	}
+	/**
+	 * This Method display the content of the Shopping Cart in Console
+	 * 
+	 */
+
+	@Override
+	public void displayCartContent() {
+		for(Item item: list){
+			System.out.println("Product: " + item.getProductName() + "   " + "Price: $" + item.getProductPrice() + ".\n");
+		}
+		
+	}
+
+	/**
+	 * This Method Calculates the total cost of all the Items in the Shopping Cart
+	 * 
+	 */
+	@Override
+	public double calculateTotalPurchase() {
+		
+		double totalCost = 0;
```

Comment:
tip double y float no son buenos para currency, si para otros cálculos matemáticos 
http://stackoverflow.com/questions/8148684/what-is-the-best-data-type-to-use-for-money-in-java-app#comment37921376_8148773